src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -4416,10 +4416,11 @@ private int convertRowIndexToModel(RowSorterEvent e, int viewIndex) {
     /**
      * Invoked when this table's <code>TableModel</code> generates
      * a <code>TableModelEvent</code>.
-     * The <code>TableModelEvent</code> should be constructed in the
-     * coordinate system of the model; the appropriate mapping to the
-     * view coordinate system is performed by this <code>JTable</code>
-     * when it receives the event.
+     * The <code>TableModelEvent</code> should be constructed in
+     * the coordinate system of the model for the column and the
+     * coordinate system of the view for the row; the appropriate
+     * mapping to the view coordinate system is performed by this
+     * <code>JTable</code> when it receives the event.
```

Comment:
A few things
- since most of the doc is changing we can update to use the more modern {@code ...} instead
- the 2nd clause is now only about the columns so should say "the appropriate mapping of the column to"
- there's no @param tag .. how come doclint isn't complaining ? But we should fix it.
- This will require a CSR.
Oh and I assume you can confirm that you verified the above is 100% true .. and there's no need to mention RowSorter - can you point to where the event is created ?

Reply:
Should all instances of <code></code>(entire JTable) changed to {@code...} or for the tableChanged method?
I was not able to understand the last point clearly. Can you please elaborate?

Reply:
I said most of the doc is changing which is clearly not most of the doc for JTable so it means just this method.

To the latter point the doc you are proposing is asserting that it does not apply to rows.
Now since we also know this is all supposed to be code called ny the implementation and responded to by the implementation and even so we are NOT going to change anything because some code somewhere would break, nontheless we want to be sure the new code is what actually happens and not just what someone wrote in a  bug report several years ago. So go find the code that calls "new TableModelEvent" and point in this PR to how it guarantees the rows are in view order.

Reply:
@prrace After going through the code, I see that the `new TableModelEvent()` is called in setModel of JTable but I wasn't able to locate the code that does model-to-view coordinate mapping for rows from here. Within the `tableChanged` method I see `sortedTableChanged` method being called if ` sortManager != null ` which in turn calls `convertRowIndexToView` method (which is called when either sorter or model changes and sorting is enabled). I believe this is the method that is responsible for mapping coordinate system from model to view for the rows.

Reply:
I was just looking at this same code too - and that's how it looks to me.
So the claim in the report AND the evaluation both look wrong to me.
But verify this with testing. I am not sure but it seems like merely doing whatever it is to install
a RowSorter will ttigger that code path and the code the submitter pointed to is only used if
the model row == the view row ..